aries_vcx/tests/test_pgsql_plugin.rs

```diff
@@ -0,0 +1,117 @@
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+#[macro_use]
+extern crate serde_json;
+
+use serde_json::Value;
+
+use aries_vcx::handlers::connection::connection::Connection;
+use aries_vcx::init::{init_issuer_config, open_as_main_wallet};
+use aries_vcx::libindy::utils::wallet::{close_main_wallet, configure_issuer_wallet, create_wallet, WalletConfig};
+use aries_vcx::settings;
+use aries_vcx::utils::devsetup::{AGENCY_DID, AGENCY_ENDPOINT, AGENCY_VERKEY};
+use aries_vcx::utils::provision::{AgentProvisionConfig, provision_cloud_agent};
+
+use crate::pgwallet::wallet_plugin::{finish_loading_postgres, load_storage_library, serialize_storage_plugin_configuration};
+
+mod pgwallet;
+
+fn _init_wallet(wallet_storage_config: &WalletStorageConfig) -> Result<(), String> {
+    info!("_init_wallet >>> wallet_storage_config:\n{}", serde_json::to_string(wallet_storage_config).unwrap());
+    match wallet_storage_config.xtype.as_ref() {
+        Some(wallet_type) => {
+            let (plugin_library_path_serialized,
+                plugin_init_function_serialized,
+                storage_config_serialized,
+                storage_credentials_serialized)
+                = serialize_storage_plugin_configuration(wallet_type,
+                                                         &wallet_storage_config.config,
+                                                         &wallet_storage_config.credentials,
+                                                         &wallet_storage_config.plugin_library_path,
+                                                         &wallet_storage_config.plugin_init_function)?;
+            let lib = load_storage_library(&plugin_library_path_serialized, &plugin_init_function_serialized)?;
+            if wallet_type == "postgres_storage" {
+                finish_loading_postgres(lib, &storage_config_serialized, &storage_credentials_serialized)?;
+            }
+            info!("Successfully loaded wallet plugin {}.", wallet_type);
+            Ok(())
+        }
+        None => {
+            info!("Using default builtin IndySDK wallets.");
+            Ok(())
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct WalletStorageConfig {
+    // Wallet storage type for agents wallets
+    #[serde(rename = "type")]
+    pub xtype: Option<String>,
+    // Optional to override default library path. Default value is determined based on value of
+    // xtype and OS
+    pub plugin_library_path: Option<String>,
+    // Optional to override default storage initialization function. Default value is  determined
+    // based on value of xtype and OS
+    pub plugin_init_function: Option<String>,
+    // Wallet storage config for agents wallets
+    pub config: Option<Value>,
+    // Wallet storage credentials for agents wallets
+    pub credentials: Option<Value>,
+}
+
+#[test]
+#[cfg(feature = "test_integration")]
+fn it_should_use_pgsql_wallet_to_provision_cloud_agent() {
```

Comment:
This kind of "it should" framework for naming tests is interesting (I have seen you do it elsewhere), but not really consistent with other test names in the repo - I would suggest something like `test_provision_cloud_agent_with_pgsql_wallet` for consistency